src/dnaio/bam.h

```diff
@@ -1,21 +1,26 @@
-#include <stdint.h>
-#include <stddef.h>
-#include <string.h>
-#include <assert.h>
-
-#ifdef __SSE2__
-#include "emmintrin.h"
+// Macros also used in htslib, very useful.
+#if defined __GNUC__
+#define GCC_AT_LEAST(major, minor) \
+    (__GNUC__ > (major) || (__GNUC__ == (major) && __GNUC_MINOR__ >= (minor)))
+#else 
+# define GCC_AT_LEAST(major, minor) 0
 #endif
 
-#ifdef __SSSE3__
-#include "tmmintrin.h"
+#ifdef __clang__
+#ifdef __has_attribute
```

Comment:
Since the above macro uses `#if defined` instead of `#ifdef`, can you do that here as well for consistency? And the two `#ifdef`s can then also combined into `if defined ... && defined`.